core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilterWithJwt.java

```diff
@@ -77,4 +89,24 @@ protected void validateAccessToken(String cookieValue)
         assertThat(claims.get("client_id")).isEqualTo(TRINO_CLIENT_ID);
         assertThat(claims.getIssuer()).isEqualTo("https://localhost:4444/");
     }
+
+    @Test
+    public void testBearerAuthenticatorUserMapping()
+            throws Exception
+    {
+        String token = hydraIdP.getToken(TRINO_CLIENT_ID, TRINO_CLIENT_SECRET, ImmutableList.of(TRINO_AUDIENCE));
+        HttpServerInfo httpServerInfo = server.getInstance(Key.get(HttpServerInfo.class));
+
+        try (Response response = httpClient.newCall(new Request.Builder()
+                        .url(uriBuilderFrom(httpServerInfo.getHttpsUri())
+                                .replacePath("/username")
+                                .toString())
+                        .addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+                        .addHeader("X-Trino-User", TRINO_CLIENT_ID)
```

Comment:
Can you please add a test with different user name for `X-Trino-User`?

Reply:
Ok will add also test when re-write is not occuring.